DEID-3212 add ecs automate sample yaml

yuya-kitani opened the pull request:
Add examples for ECS automated deployment yaml. @letmerecall @bryanbellsmith could you please review? Another documentation update PR will follow.

letmerecall commented on CloudFormation/ECS automated/cluster.yaml:
```diff
@@ -0,0 +1,267 @@
+      LaunchTemplateData: 
+        BlockDeviceMappings:
+          - Ebs:
+              VolumeSize: 50
+              VolumeType: gp3
```
Does this support some sort of `value.yaml` (or other env file) where we can keep these values?

yuya-kitani replied:
Updated!

letmerecall reviewed (commented):
@yuya-kitani I breezed over these files and they look good. However I'm wondering if we should keep a REAME to tell what to do with those settings? Like use then via portal or via cli commands. As a naive user I'd love to see the steps which I can follow.

bryanbellsmith commented on CloudFormation/ECS automated/cluster.yaml:
```diff
@@ -0,0 +1,267 @@
+                EOF
+                mkdir -p /input
+                mkdir -p /output
+                sudo chown -R ec2-user:ec2-user /input
+                sudo chown -R ec2-user:ec2-user /output
+                mount -t efs -o tls ${InputEFSId}:/ /input
+                mount -t efs -o tls ${OutputEFSId}:/ /output
```
This is interesting, I presume we're using EFS as input and output destinations for the process files via URI endpoint? Very cool

yuya-kitani replied:
Yeah that's right!

bryanbellsmith commented on CloudFormation/ECS automated/main.yaml:
```diff
@@ -0,0 +1,95 @@
+    Type: AWS::CloudFormation::Stack
+    DependsOn: VPC
+    Properties:
+      TemplateURL: https://<Your S3 bucket>/security-groups.yaml
```
[MINOR] Is this supposed to reference the https://privateai-infrastructure.s3.amazonaws.com/ecs/security-groups.yaml file?

yuya-kitani replied:
Thanks for the comment! This intends to point to customer owning s3. I will modify it to be pinpointed.

bryanbellsmith commented on CloudFormation/ECS automated/main.yaml:
```diff
@@ -0,0 +1,95 @@
+      - LoadBalancer
+      - Cluster
+    Properties:
+      TemplateURL: https://<Your S3 bucket>/service.yaml
```
[MINOR] Similarly with this URL, is this supposed to be https://privateai-infrastructure.s3.amazonaws.com/ecs/service.yaml ?

bryanbellsmith commented on CloudFormation/ECS automated/main.yaml:
```diff
@@ -0,0 +1,95 @@
+    Type: AWS::CloudFormation::Stack
+    DependsOn: SecurityGroups
+    Properties:
+      TemplateURL: https://<Your S3 bucket>/cluster.yaml
```
[MINOR] Similarly, is this supposed to be https://privateai-infrastructure.s3.amazonaws.com/ecs/cluster.yaml ?

bryanbellsmith reviewed (approved):
This is good! A few minor comments, and I'll mirror what Girish said, these files don't appear to run as-is, and need some values to be updated, etc. Perhaps a local Readme is the best approach

yuya-kitani commented:
Thanks for the feedback @letmerecall @bryanbellsmith ! I will incorporate them accordingly!

yuya-kitani commented:
@letmerecall @bryanbellsmith Thanks for the review! I added values.yaml to store variables as single customize point and added detailed README. Could you review it please?

letmerecall reviewed (approved):
LGTM!

Thanks @yuya-kitani.
